src/vs/editor/contrib/referenceSearch/referenceSearch.ts

```diff
@@ -176,12 +191,12 @@ function closeActiveReferenceSearch(accessor: ServicesAccessor, args: any) {
 		return;
 	}
 
-	controller.closeWidget();
+	fn(controller);
 }
 
 KeybindingsRegistry.registerCommandAndKeybindingRule({
 	id: 'closeReferenceSearch',
-	weight: KeybindingsRegistry.WEIGHT.editorContrib(50),
+	weight: KeybindingsRegistry.WEIGHT.workbenchContrib(50),
```

Comment:
Why workbench and not editor? This feature is also available in the monaco editor....

Reply:
@jrieken the workbench is also registering ESC for trees and lists here: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/electron-browser/commands.ts#L344. This is needed to preserve the behaviour of ESC closing the references widget. It is a bit ugly, I agree. The alternative would be to register this command also from the workbench maybe.